### ISSUE TITLE
The event is not obtained from the apiserver cache

### ISSUE THREAD

**Issue by Black-max12138** [kubernetes/kubernetes#124724]
### What happened?

When /api/v1/namespaces/sop/events is requested, data is obtained from etcd instead of the apiserver cache. When there are too many events, etcd memory increases.

### What did you expect to happen?

We expect that when /api/v1/namespaces/sop/events is requested, data is also fetched from the apiserver cache

### How can we reproduce it (as minimally and precisely as possible)?

Request /api/v1/namespaces/sop/events. Check the etcd logs. It is found that data is directly obtained from etcd instead of apiserver cache.

### Kubernetes version

<details>

```console
$ kubectl version
# 1.28
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by olderTaoist** [comment]
`/api/v1/namespaces/sop/events` is List request？if it is List request，due to without resourceVersion，the default behavior is directly obtained from etcd, you can use [ConsistentListFromCache](https://github.com/kubernetes/kubernetes/pull/118508) feature gate  change  fetching data from the apiserver cache

**Comment by Black-max12138** [comment]
> `/api/v1/namespaces/sop/events` is List request？if it is List request，due to without resourceVersion，the default behavior is directly obtained from etcd, you can use [ConsistentListFromCache](https://github.com/kubernetes/kubernetes/pull/118508) feature gate change fetching data from the apiserver cache

Yes, this is a list request, and the request carries resourceVersion, but why is it not cached?

**Comment by olderTaoist** [comment]
> > `/api/v1/namespaces/sop/events` is List request？if it is List request，due to without resourceVersion，the default behavior is directly obtained from etcd, you can use [ConsistentListFromCache](https://github.com/kubernetes/kubernetes/pull/118508) feature gate change fetching data from the apiserver cache
> 
> Yes, this is a list request, and the request carries resourceVersion, but why is it not cached?

you can refer to [shouldDelegateList](https://github.com/kubernetes/kubernetes/blob/v1.28.0/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L724),  the reason maybe is the request with resourceVersionMatch="Exact" query.
